app/src/main/java/com/puzzlebench/clean_marvel_kotlin/presentation/mvp/CharecterView.kt

```diff
@@ -50,4 +67,44 @@ class CharecterView(activity: MainActivity) {
         activityRef.get()!!.progressBar.visibility = View.VISIBLE
 
     }
+
+
+
+    fun showCharacterInformation(character: Character) {
+        val dialogFragment = CharacterDialogFragment()
+        val args = Bundle()
+        args.putParcelable(EXTRA_CHARACTER, character)
+        dialogFragment.arguments = args
+        dialogFragment.show(activityRef.get()?.supportFragmentManager, "txn_tag")
+    }
+
+    class CharacterDialogFragment() : DialogFragment() {
+
+        private lateinit var character: Character
+
+
+        override fun onCreate(savedInstanceState: Bundle?) {
+            super.onCreate(savedInstanceState)
+        }
+
+        override fun onStart() {
+            super.onStart()
+            val d: Dialog? = dialog
+            if (d != null) {
+                val width = ViewGroup.LayoutParams.MATCH_PARENT
+                val height = ViewGroup.LayoutParams.MATCH_PARENT
+                d.window?.setLayout(width, height)
+            }
+        }
+
+        override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View {
+            var rootView: View = inflater.inflate(R.layout.character_dialog_fragment, container, false)
+            character = this.arguments.getParcelable(EXTRA_CHARACTER)!!
+            rootView.imgCharacter.getImageByUrl(character.thumbnail.path + "." + character.thumbnail.extension)
```

Comment:
please try to move the dot to a constant and use like this "${character.thumbnail.path}$DOT${character.thumbnail.extension}" something like that

Reply:
Done